src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -386,7 +387,34 @@ public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilde
             );
         }
     }
-
+    public static class RenderedPolyface extends RenderedShape<ShapeDispatcher.Polyface>
+    {
+        public RenderedPolyface(Minecraft client, ShapeDispatcher.ExpiringShape shape)
+        {
+            super(client, (ShapeDispatcher.Polyface)shape);
+        }
+        @Override
+        public void renderFaces(Tesselator tessellator, BufferBuilder bufferBuilder, double cx, double cy, double cz, float partialTick)
+        {
+                bufferBuilder.begin(switch(shape.mode){
```

Comment:
you should decode the mode in the constructor so you don't have so switch every render

Reply:
done.